### PR TITLE
chore: let Sonar to scan all modules

### PR DESCRIPTION
**Pull request by algomaster99**
Sonar is not being run in the CI for the new changes in SPI. These changes try to fix that.

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -72,7 +72,7 @@ jobs:
         shell: bash
         run: .github/mem-constrained-exec.sh
       - name: Run Sonar
-        if: ${{ matrix.os == 'ubuntu-latest' && (github.event_name != 'pull_request' || github.event.pull_request.head.repo.full_name == 'SpoonLabs/sorald') }}
```
Comment (slarse) [inline .github/workflows/tests.yml:75]: That's not a good idea. You can't use secrets in pull requests from forks, so by removing this all PRs from forks will fail.

Comment (slarse) [reply .github/workflows/tests.yml:75]: The commit before you removed this safety check did run the Sonar scan, as it matches on the head repo name.

Comment (algomaster99) [reply .github/workflows/tests.yml:75]: I just did this for testing purposes. I am going to revert this eventually.

When I logged into the SonarCloud portal, I couldn't see a report for the scan made for this PR. That is why I tried to force it.

The issue has to do something with the refactoring. Until now, we only needed to register `SpoonLabs_sorald`, but now we have three modules - `sorald_parent`, `sorald`, and `sorald_api` - thanks to the #808 . The scans which I saw throw warnings that they are not able to find the sorald-api JAR.

**Comment by algomaster99** [comment]
We are getting somewhere.  We have the [report on SonarCloud](https://sonarcloud.io/code?id=SpoonLabs_sorald&pullRequest=811) for this PR even though it does not scan the sub modules. Trying to figure out what is wrong.

**Comment by algomaster99** [comment]
Not asking anyone to review this because:
1. I am a bit confident that it should work
2. There is an urgent need to merge #808 because some features rely on it. This PR is a prerequisite for it.
3. Even if the Sonar scan fails on `master`, I can fix it later because it is not a big issue.

**Pull request merged**